### ISSUE TITLE
Sort by Composer

### ISSUE THREAD

**Issue by p5a0u9l** [cmus/cmus#612]
How reasonable is it to request a feature that allows better support for music with extended tags, most importantly, Composer? One in particular would be extending `aaa_mode` to `aaac_mode` (which of course breaks the `aaa` symmetry, but you see the idea hopefully). 

If similar requests (support for classical music tags, etc.) has previously come up in forums, pardon and point me there. 

To sweeten the deal, I'd be happy to contribute said support where possible. I'm familiar with `c`, but less so with contributing to large open-source projects, but have been looking for a good opportunity to dive in somewhere.

**Comment by nefthy** [comment]
> To sweeten the deal, I'd be happy to contribute said support where possible. I'm familiar with c, but less so with contributing to large open-source projects, but have been looking for a good opportunity to dive in somewhere.

cmus is quite welcoming contributions. Just clone the repository, hack away and make a pull request when your done. However keep the following in mind:

- Use and keep the formatting (linux kernel convetions) and style (c99), that is used in the source files.
- Don't break current behavior if possible. If this is not the case, discuss it with the maintainers (@flyingmutant , @mahkoh) first.

**Comment by mahkoh** [comment]
Before writing any code you should explain in more detail what changes you want.

**Comment by p5a0u9l** [comment]
@mahkoh agreed completely. If I can communicate the functionality I'm after sufficiently, you will be
better equipped to point me to the relevant places in the code as well as advise on the feasibility of the changes. 

The general theme is increased support for classical music tags in cmus.  Here's a simple proposed starting point, 

Add the "composer" tag to the list of Filter strings, which, from the man page, currently are limited 
to 
     filename, artist, albumartist, album, title, genre, comment, codec, codec_profile, media

this would enable creating a filter like, 
    composer="*Bach*"&artist="*Perahia"

More ambitious functionality would add Composer to view 1 in addition to Artist/Album. Basically, allow sort by Composer OR Artist OR Album. But, how's starting with the first idea, where in the code should I look? Cheers

**Comment by paulrbts** [comment]
I'd definitely support/use this! I don't know C so can't help write it but if I can help test it, I'd gladly help.

**Comment by jkarres** [comment]
I've just created a pull request (#666) that creates a setting enabling sorting the tree view by composer.  (If there's a way to associate it to this issue, I'm afraid I didn't see it.)  It's still got issues, but I'd appreciate it if anybody could tell me if I'm on the right path.

**Comment by h4fnp** [comment]
> Before writing any code you should explain in more detail what changes you want.

The title of this feature request is a bit misleading, I think. I propose to change it to: **"Add support for additional ID3v2 tags (Language, Composer, Label, Mood etc.)"**

## List of fields

Here is a list of the fields that have already been mentioned / would IMHO be beneficial to be supported (i.e. can be used with `:filter`, `:fset`, or `:mark` expressions, as well as in format strings via `%{tagname}`):

| Filter key | ID3v2 frame | ID3v2 spec | Vorbis comment | Description |
|---|---|---|---|---|
| `language` | `TLAN` | [ID3v2.3](https://id3.org/id3v2.3.0#TLAN) | `LANGUAGE` | ISO 639-2 three-letter language code(s) of the audio content (e.g. `eng`, `deu`, `jpn`) |
| `composer` | `TCOM` | [ID3v2.3](https://id3.org/id3v2.3.0#TCOM) | `COMPOSER` | Composer(s) of the work |
| `label` | `TPUB` | [ID3v2.3](https://id3.org/id3v2.3.0#TPUB) | `LABEL` | Record label or publisher |
| `mood` | `TMOO` | [ID3v2.4](https://id3.org/id3v2.4.0-frames) | `MOOD` | Free-text mood keywords (e.g. `Sad`, `Energetic`) |

Vorbis comment field names are taken from the [MusicBrainz Picard tag map](https://picard-docs.musicbrainz.org/downloads/MusicBrainz_Picard_Tag_Map.html).

Note: `TMOO` is technically ID3v2.4 only, but is widely written into ID3v2.3 files in practice by tools such as MusicBrainz Picard and should be safe to read from either version.

## Current behavior

Attempting to filter on any of these fields currently yields:

```
Error: error parsing filter: unknown key language
```

The same error occurs for `composer`, `label`, and `mood`.

## Expected behavior

After this change, the new fields should be usable in three places:

**Filters** (`:filter`, `:fset`, `:mark`):
```
:filter language=eng
:fset classical=composer~Bach
:fset label=ECM
:filter mood~Melancholic
```

**Format strings** (`format_current`, `format_playlist`, `format_trackwin`, etc.) via the `%{tagname}` syntax:
```
set format_current= %{?composer?%{composer} - }%{title} %{?language?[%{language}]}%= %{duration}
```

**Format string conditionals**, since their `CONDITION` syntax is shared with filters:
```
set format_current= %{?language=eng?%{title}?%{title} (%{language})}%= %{duration}
```
